Keyutils-cifs

zpytela opened the pull request.

WOnder93 commented:
I would prefer to have the `cifs.upcall` stuff in a separate new policy module (called `cifs_utils`), since both `/usr/sbin/cifs.upcall` and `/etc/request-key.d/cifs.spnego.conf` (which registers it with the request-key dispatcher) are shipped by `cifs-utils`, which is independent from `keyutils`.

WOnder93 commented:
Also, I wonder if using a separate new type for `/usr/sbin/key.dns_resolver` wouldn't be better. I suspect it will do a different set of things from `/usr/sbin/request-key` and by principle of least privilege it would be better to confine each separately. (But if it is shown that it does only a subset of what we already have for `keyutils_request_t`, then it can be fine to just add [the missing `execute_no_trans`](https://bugzilla.redhat.com/show_bug.cgi?id=2182643#c9) and do it this way.)

zpytela committed:
Label key.dns_resolver with keyutils_dns_resolver_exec_t

Label /usr/sbin/key.dns_resolver with keyutils_dns_resolver_exec_t,
allow request-key execute this file with a transition,
add initial policy for the keyutils_dns_resolver_t domain.

The commit addresses issues starting with the following AVC denial:
type=AVC msg=audit(1683047200.533:218): avc:  denied  { execute } for  pid=5107 comm="request-key" name="key.dns_resolver" dev="dm-0" ino=1061417 scontext=system_u:system_r:keyutils_request_t:s0 tcontext=system_u:object_r:bin_t:s0 tclass=file permissive=0

Resolves: rhbz#2182643

zpytela committed:
Add initial policy for cifs-helper

Label /usr/sbin/cifs.upcall with cifs_helper_exec_t,
allow request-key execute this file with a transition,
add initial policy for the cifs_helper_t domain.
Add cifs policy module.

The commit addresses the following AVC denial:
May 07 13:48:02 hostname audit[1396]: AVC avc:  denied  { execute } for  pid=1396 comm="request-key" name="cifs.upcall" dev="sda4" ino=3836882 scontext=system_u:system_r:keyutils_request_t:s0 tcontext=system_u:object_r:bin_t:s0 tclass=file permissive=0

Resolves: rhbz#2182643
